cruise-control-metrics-reporter/src/test/java/com/linkedin/kafka/cruisecontrol/metricsreporter/utils/CCEmbeddedBroker.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright 2018 LinkedIn Corp. Licensed under the BSD 2-Clause License (the "License").  See License in the project root for license information.
+ */
+
+package com.linkedin.kafka.cruisecontrol.metricsreporter.utils;
+
+import java.io.File;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Method;
+import java.net.URI;
+import java.util.HashMap;
+import java.util.Map;
+import org.apache.commons.io.FileUtils;
+import org.apache.kafka.common.network.ListenerName;
+import org.apache.kafka.common.security.auth.SecurityProtocol;
+import org.apache.kafka.common.utils.Time;
+
+
+public class CCEmbeddedBroker implements AutoCloseable {
+  private final static Class<?> KAFKA_CONFIG_CLASS;
+  private final static Class<?> KAFKA_SERVER_CLASS;
+  private final static Class<?> SCALA_OPTION_CLASS;
+  private final static Class<?> SCALA_SEQ_CLASS;
+  private final static Class<?> SCALA_ARRAY_BUFFER_CLASS;
+  private final static Constructor<?> CONFIG_CTR;
+  private final static Constructor<?> SERVER_CTR;
+  private final static Constructor<?> ARR_BUF_CTR;
+  private final static Method STARTUP_METHOD;
+  private final static Method SHUTDOWN_METHOD;
+  private final static Method AWAIT_SHUTDOWN_METHOD;
+  private final static Object EMPTY_OPTION;
+  private final static Method BOUND_PORT_METHOD;
```

Comment:
What's the reason for using reflection here?

Reply:
Originally (i.e. in `master`) Cruise Control (CC) used to depend on the test harness of `li-apache-kafka-clients`. In `migrate_to_kafka_2_0` branch we extracted this test harness from `li-apache-kafka-clients`. While using reflection helps `li-apache-kafka-clients` avoid having to depend on a particular scala version, this does not apply to CC; hence, there is no reason to use reflection here.
We should refactor this part, thanks for the feedback!

Reply:
Thanks for the explanation.